test/install_script/nvm_check_global_modules

```diff
@@ -6,16 +6,21 @@ cleanup () {
 
   rm -f npm
   PATH="$ORIGINAL_PATH"
+  NVM_DIR="$ORIGINAL_NVM_DIR"
 
   unset -f setup cleanup die
-  unset message ORIGINAL_PATH
+  unset message ORIGINAL_PATH ORIGINAL_NVM_DIR
 }
 die () { echo "$@" ; cleanup ; exit 1; }
 
 NVM_ENV=testing \. ../../install.sh
 
 setup () {
   ORIGINAL_PATH="$PATH"
+  ORIGINAL_NVM_DIR="$NVM_DIR"
+
+  # Pretend we're not using NVM
+  unset NVM_DIR
```

Comment:
this seems like it would make line 33 require a system node, which won’t be present when we migrate from Travis to GitHub actions. Can this line be moved until just before the tests that need it?

Reply:
I don't believe it does — I ran the test in a Docker container with no system node, and added some checks to make sure `command -v npm` consistently returned the same value (except for the last test, which alters the `PATH` on line 55), and it still passed.

I think because we don't (except in that one test) change the `PATH`, it's still using the `PATH` set in the [`Makefile`](https://github.com/dmolesUC/nvm/blob/e3823f65ca9ac50759dc4eec83e8ffe0d7c1047d/Makefile#L11), and so still using the node installed under `nvm-sh/nvm/versions`.

Reply:
The first test, though, runs `npm install -g nop` - what happens if there is no system node, and no other nvm-managed node?

line 16 above will ensure nvm is installed, but that doesn't actually install a node version.

Reply:
In order to get as far as running this test, there already has to be an installed node on the `PATH` (the Travis build does this [here](https://github.com/nvm-sh/nvm/blob/510b2fd44c79049e16c5ff06988de62526504d0b/.travis.yml#L22-L23)), and as far as I can tell, in practice, the node on the `PATH` is always the one installed/managed by nvm, whether or not there's another node on the system.

Clearing `NVM_DIR` doesn't change the `PATH`, and node itself (of course) doesn't know or care about `NVM_DIR`, so `npm install` and `npm uninstall` continue to work just fine.

We could refactor the test to do everything in subshells, and make it more clear what's going on, but I'm not sure it's worth it.